App.test.js

```diff
@@ -0,0 +1,35 @@
+import { render, fireEvent, waitFor } from '@testing-library/react';
+import App from './App';
+import { loginUser } from './api';
+import { BrowserRouter as Router } from 'react-router-dom';
+
+jest.mock('./api');
+
+test('renders login page', () => {
+  const { getByText } = render(<App />);
+  const linkElement = getByText(/Login/i);
+  expect(linkElement).toBeInTheDocument();
+});
+
+test('logs in successfully', async () => {
+  const user = { email: 'test@test.com', password: 'password' };
+  loginUser.mockResolvedValue(user);
+
+  const { getByLabelText, getByText } = render(
+    <Router>
+      <App />
```

Comment:
Consider adding error handling in your tests. If the login fails, the test should be able to catch and handle the error.